backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -245,6 +253,13 @@ private Project insertProject(Project project, String workspaceName, UserInfo us
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
       Workspace workspace = roleService.getWorkspaceByWorkspaceName(userInfo, workspaceName);
 
+      Query deletedEntitiesQuery = session.createQuery(GET_DELETED_PROJECTS_IDS_BY_NAME_HQL);
+      deletedEntitiesQuery.setParameter("projectName", project.getName());
+      List<String> deletedEntityIds = deletedEntitiesQuery.list();
+      if (!deletedEntityIds.isEmpty()) {
+        roleService.deleteEntityResources(deletedEntityIds, ModelDBServiceResourceTypes.PROJECT);
```

Comment:
I just want to point out that the fact that our cleanup is idempotent means that we can do this part of the cleanup earlier without worrying about issues downstream.